lib/std/json/static.zig

```diff
@@ -250,7 +263,10 @@ fn internalParse(
             var name_token: ?Token = try source.nextAllocMax(allocator, .alloc_if_needed, options.max_value_len.?);
             const field_name = switch (name_token.?) {
                 inline .string, .allocated_string => |slice| slice,
-                else => return error.UnexpectedToken,
+                else => {
+                    freeAllocated(allocator, name_token.?);
```

Comment:
do you have a usecase where this deallocation is meaningful or observable?
* If you're not using a `*Leaky` parse function, then the entire arena gets deinitted on error.
* If you are using a `*Leaky` parse function, there's more than just this allocation that's getting leaked on error, such as every string value parsed before this point.

I understand that this just "makes more sense" to cleanup after yourself, but one of the joys of using an arena allocator is not having to think about this kind of thing.

Reply:
I believe it would be as meaningful and observable as you described. I can remove this `freeAllocated()` call, as well as the analogous one in the `.Union` prong's `field_name` `switch`, I'll push that once the below comment is resolved. I believe the other `freeAllocated()` calls that were added are still appropriate; the one in an `.object_end` prong doesn't trigger an error, and therefore doesn't trigger an `errdefer`, and the one in the `inline for`'s `else` branch is analogous to the already existing one in the `.Struct` prong.